crypto/x509/x509_vfy.c

```diff
@@ -3295,7 +3290,7 @@ static int build_chain(X509_STORE_CTX *ctx)
         CB_FAIL_IF(DANETLS_ENABLED(dane)
                        && (!DANETLS_HAS_PKIX(dane) || dane->pdpth >= 0),
                    ctx, NULL, num - 1, X509_V_ERR_DANE_NO_MATCH);
-        if (X509_self_signed(curr, 0))
+        if (X509_self_signed(sk_X509_value(ctx->chain, num -1), 0))
```

Comment:
missing space after `-`

Reply:
Approved with this nit fixed.

Reply:
Thx, fixed.

And made a further simplification a little thereafter, 
replacing `sk_X509_num(ctx->chain)` by `num`.